src/main/java/de/fraunhofer/aisec/cpg/passes/CallResolver.java

```diff
@@ -276,7 +276,7 @@ private void resolveArguments(CallExpression call, RecordDeclaration curClass) {
   }
 
   private void handleNormalCalls(RecordDeclaration curClass, CallExpression call) {
-    if (curClass == null && this.currentTU != null) {
+    if (curClass == null && this.currentTU != null && !(call instanceof MemberCallExpression)) {
```

Comment:
I would actually prefer, if we would call `handleMethodCall` in the if-branch of the `handleCallExpression` function, something like this:

```
   if (call instanceof MemberCallExpression) {
      Node member = ((MemberCallExpression) call).getMember();
      if (member instanceof HasType
          && ((HasType) member).getType() instanceof FunctionPointerType) {
        handleFunctionPointerCall(call, member);
        return;
      } else {
        handleMethodCall(curClass, call);
        return;
      }
    }
```

I think that would be cleaner, since there is already this if, that decides how to proceed with member call expressions, instead of burying the decision in `handleNormalCalls`.